### ISSUE TITLE
unit_value=0 causes NaN on prices per unit

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#7322]
## Description

If unit_value is set to zero - this can be done through the UI - then unit prices in the shopfront will be displayed as NaN / (unit); But shown correctly in the BO.

## Expected Behavior

NaN should not be visible to the customer; unit prices should not show any info in these cases OR unit price info in the shopfront should match the info seen by the admin.

## Actual Behaviour

NaN is currently visible to the customer; there are discrepancies between shopfront and BO in these cases.

## Steps to Reproduce

1. Create an item product, via `/admin/products/new`
2. Set "Value" = 0
3. Finish creating the product and add it to an open OC.
4. Edit the newly created variant and check; see the unit price info.
5. As a customer, visit the shopfront, and see the unit price info -> notice the NaN

## Animated Gif/Screenshot

![image.png](https://images.zenhubusercontent.com/5d761b049dde650001eaf32f/3595ceb2-4493-49db-afc3-0b14a0d18666)

## Workaround

Currently under feature toggle:
No workaround as a customer; does not interfere with function (apparently).

## Severity

bug-s3?
Not sure we don't need more investigation, something else might be breaking.

## Your Environment

* Version used: v3.6.2 - currently under feature toggle.
* Browser name and version: Firefox 86
* Operating System and version (desktop or mobile): Desktop Ubuntu 20.04

## Possible Fix

It was noticed that `unit_value` cannot be blank and is automatically set to 1, if left blank in the UI (#6369). However, `unit_value` does accepts value zero which, together with `weight=null`may be breaking the UI somewhere else, and causing bugs related to those observed before #6527 (I'm not sure - keeping an eye on this though)

**Comment by jibees** [comment]
Hi @filipefurtad0 
Thanks for catching this one ;)
Just to clarify, when i create a new product, and set its value to `0` (which at some point can be strange, but anyway), I can't have the field `Unit Price` to be filled. It's empty, as you can see here:

<img width="1043" alt="Capture d’écran 2021-04-06 à 21 17 43" src="https://user-images.githubusercontent.com/296452/113766538-d2b73580-971d-11eb-8e93-b9681a0c86c1.png">

On the other hand, if I edit a variant and set its `value` to `0` inside the Bulk Edit Product page: 
<img width="852" alt="Capture d’écran 2021-04-06 à 21 22 52" src="https://user-images.githubusercontent.com/296452/113766893-48230600-971e-11eb-944f-ca8fcb045564.png">

As a result, I can see a `NaN` in the shopfront:
<img width="683" alt="Capture d’écran 2021-04-06 à 21 21 50" src="https://user-images.githubusercontent.com/296452/113766957-5a04a900-971e-11eb-9e3b-7b1715e6dd9e.png">

**TLDR;** I can reproduce `NaN` in the shopfront, but `Unit Price` field in the shop front is empty.

**Comment by jibees** [comment]
Maybe @andrewpbrett could be really helpful on this one ;)
It happens when `unit_price.denomibator` is equal to `0.0` then we have a `null` value in the json: 
```json
"unit_price_price":null
```
(which is pretty logical as we try to divide by zero (`price_with_fees / (unit_price.denominator || 1)`)
I'm wondering if I should handle that inside the js (and don't display the unit price like it is done in the backoffice), or maybe there is another solution.
What do you think @andrewpbrett ? 🙏

**Comment by jibees** [comment]
In the past order, (using `app/models/spree/line_item.rb#209`) we _show unit price with null `value`_ in a different way:
<img width="1002" alt="Capture d’écran 2021-04-06 à 22 28 55" src="https://user-images.githubusercontent.com/296452/113774487-a43e5800-9727-11eb-8deb-9f8f87039a69.png">

**Comment by jibees** [comment]
@andrewpbrett I've drafted a PR, #7331 that I would like you to take a look at 🙏

**Comment by filipefurtad0** [comment]
Thanks for looking at this one and reproducing it @jibees :+1: 

Yeah, I realize this is a bit of an edge case (manually typing `0` as `unit_value`). On other places on teh app (ex. product import) I think it is easier for the user to be less aware and actually do this by mistake.

Also (and possibly not relevant for this issue and PR), when a product is created as `items` it will have `weight=null`; changing this item product to a weight product does not fill in `weight` with zero, it remains `null` for the master variant and those additionally created.
